### PR TITLE
Optimize Canvas loading

### PR DESCRIPTION
**Pull request by alstojanovic**
### Describe the background of your pull request

What does your pull request do? Does it solve a bug (which one?), add a feature?

### Additional context

Add any other context

### Governance 

- [ ] Documentation is added
- [ ] Test cases are added or updated

By submitting this pull request I confirm that:
- I've read the contributing document https://github.com/alkem-io/.github/blob/master/CONTRIBUTING.md.
- I've read and understand the Code of Conduct https://github.com/alkem-io/.github/blob/master/CODE_OF_CONDUCT.md.
- I understand that any contributions or suggestions I made may make it into the actual code. I've read the License 
     https://github.com/alkem-io/Coordination/blob/master/LICENSE.
- I understand that submitting the pull request requires agreeing to and signing the contributor license agreement.

### CODE REVIEW

File: src/domain/collaboration/canvas/containers/canvasQueries.graphql [hunk 1]
```diff
@@ -108,6 +134,22 @@ query hubCanvasValues($hubId: UUID_NAMEID!, $calloutId: UUID_NAMEID!, $canvasId:
...
+    id
+    challenge(ID: $challengeId) {
+      id
+      collaboration {
+        ...CalloutWithCanvas
+      }
+    }
+  }
+}
 query challengeCanvases($hubId: UUID_NAMEID!, $challengeId: UUID_NAMEID!) {
```
Comment (ccanos) [inline src/domain/collaboration/canvas/containers/canvasQueries.graphql:153]: Can you remove these if we are not using them anymore? I think we are still using hubCanvasesLazyQuery in the admin section and I'm not sure how hard would it be to use these new ones, but I think for Challenges and Opportunities we don't use them anymore

File: src/domain/collaboration/canvas/containers/CanvasProvider.tsx [hunk 2]
```diff
@@ -22,7 +22,7 @@ export type TemplateQuery = {
 } & { hubId: string };
 
 export interface IProvidedEntities {
-  canvases: CanvasFragmentWithCallout[];
+  canvas: CanvasDetailsFragment | undefined;
   templates: CreateCanvasCanvasTemplateFragment[];
   calloutId: string | undefined;
   authorization: NonNullable<CollaborationWithCanvasDetailsFragment['callouts']>[0]['authorization'];
```
Comment (ccanos) [inline src/domain/collaboration/canvas/containers/CanvasProvider.tsx:28]: I think, if you get to get rid of [hub/ch/opp]Canvases queries (see the other comment) you'll be able to remove CollaborationWithCanvasDetails fragment and this one can become `CanvasDetailsFragment['authorization']` 
** I think **

**Review by ccanos — CHANGES_REQUESTED** [review]
Just a few things to clean up if possible, the logic looks good to me and the code works

File: src/domain/collaboration/canvas/containers/CanvasProvider.tsx (hunk 2 above)
Comment (alstojanovic) [reply src/domain/collaboration/canvas/containers/CanvasProvider.tsx:28]: I cannot remove HubCanvases, it is still used and it uses CollaborationWithCanvasDetails.

File: src/domain/collaboration/canvas/containers/canvasQueries.graphql (hunk 1 above)
Comment (alstojanovic) [reply src/domain/collaboration/canvas/containers/canvasQueries.graphql:153]: Yes, removed challengeCanvases and opportunityCanvases queries.

**Review by ccanos — APPROVED** [review]

**Pull request merged**